cmd/shortener/main.go

```diff
@@ -1,3 +1,114 @@
 package main
 
-func main() {}
+import (
+	"github.com/go-chi/chi/v5"
+	"github.com/sub3er0/urlShorteningService/cmd/config"
+	"io"
+	"math/rand"
+	"net/http"
+	"net/url"
+)
+
+type URLShortener struct {
+	urls          map[string]string
+	ServerAddress string
+	BaseURL       string
+}
+
+func (us *URLShortener) GetHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		http.Error(w, "Only GET requests are allowed!", http.StatusBadRequest)
+		return
+	}
+
+	id := r.PathValue("id")
+
+	for k, v := range us.urls {
+		if k == id {
+			w.Header().Set("Location", v)
+			w.WriteHeader(http.StatusTemporaryRedirect)
+			return
+		}
+	}
+}
+
+func (us *URLShortener) PostHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Error(w, "Only POST requests are allowed!", http.StatusBadRequest)
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
Не стоит забывать закрывать тело запроса после чтения.

```suggestion
	body, err := io.ReadAll(r.Body)

	if err != nil {
		http.Error(w, err.Error(), http.StatusBadRequest)
		return
	}
	defer r.Body.Close()
```

Reply:
исправлено